### PR TITLE
feat: updated README instructions for MVP

### PR DESCRIPTION
**Pull request by dchoi27**
This needs eyes especially in two areas:
- Do these getting started docs make enough sense for someone to pick up and use? There are some sections marked coming soon but think that's fine for now
- Are the instructions for registering with NFT.Storage in the intro + in the client lib docs right?

Partially closes https://github.com/web3-storage/w3up-client/issues/84

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,6 +164,24 @@ In the example above, `directoryCid` resolves to an IPFS directory with the foll
...
 ```
+### Alternate implementation options
+
+As discussed above, there are options outside the traditional client-server model that w3up supports. We how to use `w3up-client` to achieve these options in this section.
+
+#### Direct upload from client
+
+In this option, you (the developer) own your Space, but delegate permissions to your users to directly upload content to the service on your behalf. This isn’t completely “serverless” - you still need some infrastructure to create delegated UCAN tokens, but it’s minimal, and potentially saves a ton of bandwidth and overhead.
+
+- _To-do: Include instructions_
```
Comment (dchoi27) [inline README.md:175]: Something like:
- First follow the instructions in [https://github.com/web3-storage/w3up-client#creating-a-client-object](https://github.com/web3-storage/w3up-client#creating-a-client-object), [https://github.com/web3-storage/w3up-client#creating-and-registering-spaces](https://github.com/web3-storage/w3up-client#creating-and-registering-spaces)
- Net new step: Delegate
    - Can have your user’s `w3up-client` instance create a Space keypair using the client’s `createSpace` ([https://github.com/web3-storage/w3up-client#createspace](https://github.com/web3-storage/w3up-client#createspace))
        - You can persist this Space keypair in the client’s state (e.g., store the keypair in your backend user database and send it back to your user when they login) or generate a new Space for each session
    - Your user should make a request to you when they want to upload something, and you can create a UCAN delegating upload permission to that Space DID: [https://github.com/web3-storage/w3up-client#createdelegation](https://github.com/web3-storage/w3up-client#createdelegation)
    - From there, they can use that UCAN token to upload content directly to the service by using [https://github.com/web3-storage/w3up-client#addproof](https://github.com/web3-storage/w3up-client#addproof) and one of the upload methods (*****************************************once the proof is added, does the client know to use it for the upload methods?*****************************************)
    - If you want to know what upload each user made, you should have them make a request to you in parallel with the upload’s CID and other info you want to know about their upload
- Then follow the instructions to upload here: [https://github.com/web3-storage/w3up-client#uploading-data](https://github.com/web3-storage/w3up-client#uploading-data)

File: README.md [hunk 2]
```diff
@@ -66,6 +67,23 @@ Agents and Spaces are both generated by `w3up-client` on the user's local machin
 
 This section shows some of the basic operations available in the `w3up-client` package. See the [API reference docs][docs] or the source code of the [`w3up-cli` package][w3up-cli-github], which uses `w3up-client` throughout.
 
+Before data can be uploaded via the client, the client needs to have permissions to upload to the target service either by having permissions to a registered `Space`. This means either you (the developer) or your user needs to have a registered Space.
```
Comment (gobengo) [inline README.md:70]:
```suggestion
Before data can be uploaded via the client, the client needs to have authorization to upload to the target service either by having permissions to a registered `Space`. This means either you (the developer) or your user needs to have a registered Space.
```

File: README.md (hunk 1 above)
Comment (dchoi27) [reply README.md]: just saying coming soon for now, saving these notes for later

File: README.md [hunk 3]
```diff
@@ -114,7 +134,15 @@ try {
 
 Calling `registerSpace` will cause an email to be sent to the given address. Once a user clicks the confirmation link in the email, the `registerSpace` method will resolve. Make sure to check for errors, as `registerSpace` will fail if the email is not confirmed within the expiration timeout.
 
-Registering a space enrolls it in web3.storage's free usage tier, allowing you to store files, list uploads, etc.
+By default, calling `registerSpace` registers the Space with web3.storage w3up. You can pass the optional `provider` param to register with NFT.Storage w3up instead.
+
+```js
+try {
+  await client.registerSpace('zaphod@beeblebrox.galaxy', provider = 'did:web:nft.storage')
```
Comment (alanshaw) [inline README.md:141]:
```suggestion
  await client.registerSpace('zaphod@beeblebrox.galaxy', { provider: 'did:web:nft.storage' })
```

**Review by alanshaw — APPROVED** [review]
I think in general there is too much information on this README. We should have short and most important info here and more in-depth guides in another file linked from here.

**Pull request merged**